python/chemiscope/structures/__init__.py

```diff
@@ -28,9 +41,16 @@ def _guess_adapter(frames):
     frames as a list and a string describing which adapter should be used.
     """
 
-    ase_frames, use_ase = _ase_valid_structures(frames)
-    if use_ase:
-        return ase_frames, "ASE"
+    try:
+        ase_frames, use_ase = _ase_valid_structures(frames)
+        if use_ase:
+            return ase_frames, "ASE"
+    except TypeError:
```

Comment:
I think the TypeError should be caught in `_ase_valid_structures`, and then make the function return `[], False` in this case.

Reply:
Done.